coordinator/src/node/rollover.rs

```diff
@@ -156,14 +156,18 @@ impl Rollover {
 impl Node {
     async fn check_if_eligible_for_rollover(
         &self,
-        conn: &mut PgConnection,
+        pool: Pool<ConnectionManager<PgConnection>>,
         notifier: mpsc::Sender<OrderbookMessage>,
         trader_id: PublicKey,
         network: Network,
     ) -> Result<()> {
+        let mut conn = spawn_blocking(move || pool.get())
+            .await
+            .expect("task to complete")?;
```

Comment:
Why do we spawn_blocking here but not elsewhere? Additionally, can we not use a native async connection pool if this is a big issue?

Reply:
> Why do we spawn_blocking here but not elsewhere?

What do you mean by "elsewhere"? In this PR I have tried to consistently use `spawn_blocking` when calling `pool.get()`. 
It's very possible that we aren't doing this consistently, but we should because `get` blocks until the connection is retrieved or until a timeout expires.

> Additionally, can we not use a native async connection pool if this is a big issue?

Yep, we can, although we didn't prioritise it [thus far](https://github.com/get10101/10101/issues/142#issuecomment-1436169522). It looks like we could use https://github.com/djc/bb8.

Reply:
> What do you mean by "elsewhere"? In this PR I have tried to consistently use spawn_blocking when calling pool.get().

My mistake, maybe I was checking the deleted and not added part :sweat_smile: